src/components/app/ProfileFilterNavigator.js

```diff
@@ -4,21 +4,70 @@
 
 // @flow
 
+import React from 'react';
 import explicitConnect from '../../utils/connect';
 import { popCommittedRanges } from '../../actions/profile-view';
-import { getPreviewSelection } from '../../selectors/profile';
+import {
+  getPreviewSelection,
+  getProfileFilterPageData,
+  getProfileRootRange,
+} from '../../selectors/profile';
 import { getCommittedRangeLabels } from '../../selectors/url-state';
 import { getFormattedTimeLength } from '../../profile-logic/committed-ranges';
 import FilterNavigatorBar from '../shared/FilterNavigatorBar';
+import Icon from '../shared/Icon';
 
 import type { ElementProps } from 'react';
+import type { ProfileFilterPageData } from '../../types/profile-derived';
+import type { StartEndRange } from '../../types/units';
 
-type Props = ElementProps<typeof FilterNavigatorBar>;
+type Props = {|
+  +filterPageData: ProfileFilterPageData | null,
+  +rootRange: StartEndRange,
+  ...ElementProps<typeof FilterNavigatorBar>,
+|};
 type DispatchProps = {|
   +onPop: $PropertyType<Props, 'onPop'>,
 |};
 type StateProps = $ReadOnly<$Exact<$Diff<Props, DispatchProps>>>;
 
+class ProfileFilterNavigatorBar extends React.PureComponent<Props> {
+  render() {
+    const {
+      className,
+      items,
+      selectedItem,
+      uncommittedItem,
+      onPop,
+      filterPageData,
+      rootRange,
+    } = this.props;
+
+    let firstItem;
+    if (filterPageData) {
+      firstItem = (
+        <>
+          <Icon iconUrl={filterPageData.favicon} />
+          {filterPageData.hostname} (
+          {getFormattedTimeLength(rootRange.end - rootRange.start)})
+        </>
+      );
+    } else {
+      firstItem = 'Full Range';
+    }
+
+    return (
+      <FilterNavigatorBar
+        className={className}
+        items={[firstItem, ...items]}
```

Comment:
This is passing in a new array each time and will invalidate any PureComponent caching on the FilterNavigatorBar. I could see a fix being either to memoize the creation of the array, or pass in an addition "firstItem" prop to `FilterNavigatorBar`. Check out the usage of `memoize-immutable` in the project for memoizing examples.